### PR TITLE
let standalone npd use kubelet credentials

### PR DESCRIPTION
**Pull request by dekkagaijin**
This will allow for the NODE_PROBLEM_DETECTOR_TOKEN to be removed from GCE metadata, while retaining the increased visibility offered by a `standalone` NPD.

/kind feature
/kind cleanup
/sig cloud-provider
/area provider/gcp
/release-note-none
/priority important-soon

```release-note
NONE
```

### CODE REVIEW

File: cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: kubelet-user-npd-binding
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:node-problem-detector
```
Comment (awly) [inline cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml]: where is this role defined?

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1245,6 +1245,12 @@ current-context: service-account-context
 EOF
 }
 
+function setup-kubelet-user-npd-creds {
```
Comment (awly) [inline cluster/gce/gci/configure-helper.sh]: how about `create-node-problem-detector-kubeconfig-from-kubelet`

File: cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml (hunk 1 above)
Comment (dekkagaijin) [reply cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml]: in the parent directory's npd.yaml

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (dekkagaijin) [reply cluster/gce/gci/configure-helper.sh]: done

**Comment by dekkagaijin** [comment]
@cheftako any concerns? would be helpful to get this merged in 1.17

**Comment by dekkagaijin** [comment]
Ping @wangzhen127 @cheftako

File: cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: User
+  name: kubelet
```
Comment (mikedanese) [inline cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml]: Why is this needed? What did it fix?

File: cluster/gce/gci/configure-helper.sh [hunk 4]
```diff
@@ -1245,6 +1245,12 @@ current-context: service-account-context
 EOF
 }
 
+function create-node-problem-detector-kubeconfig-from-kubelet {
+  echo "Creating node-problem-detector kubeconfig from /var/lib/kubelet/kubeconfig"
+  mkdir -p /var/lib/node-problem-detector
+  cp /var/lib/kubelet/kubeconfig /var/lib/node-problem-detector/kubeconfig
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: Why copy? Why not just configure NPD to read the kubelet's kubeconfig directly?

File: cluster/gce/gci/configure-helper.sh [hunk 5]
```diff
@@ -2781,7 +2790,14 @@ function main() {
       create-kubeproxy-user-kubeconfig
     fi
     if [[ "${ENABLE_NODE_PROBLEM_DETECTOR:-}" == "standalone" ]]; then
-      create-node-problem-detector-kubeconfig ${KUBERNETES_MASTER_NAME}
+      if [[ -n "${NODE_PROBLEM_DETECTOR_TOKEN:-}" ]]; then
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: I'm pro just removing NODE_PROBLEM_DETECTOR_TOKEN in one swoosh and not supporting both.

Comment (dekkagaijin) [reply cluster/gce/gci/configure-helper.sh]: IIUC that'd break this script on plain GCE

File: cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml (hunk 3 above)
Comment (dekkagaijin) [reply cluster/addons/node-problem-detector/kubelet-user-standalone/npd-binding.yaml]: It makes sure that the kubelet user has all the permissions that NPD requires

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (dekkagaijin) [reply cluster/gce/gci/configure-helper.sh]: I figured that it'd be a larger, more confusing/fragile change. This way the used kubeconfig is always in the same location, regardless of NPD configuration
